packages/playground/website/src/components/ensure-playground-site/ensure-playground-site-is-selected.tsx

```diff
@@ -0,0 +1,71 @@
+import { useEffect } from 'react';
+import { resolveBlueprint } from '../../lib/resolve-blueprint';
+import { useCurrentUrl, useSearchParams } from '../../lib/router-hooks';
+import {
+	addSite,
+	setActiveSite,
+	useAppDispatch,
+	useAppSelector,
+} from '../../lib/redux-store';
+import {
+	createNewSiteInfo,
+	getSiteInfoBySlug,
+	SiteStorageType,
+} from '../../lib/site-storage';
+
+/**
+ * Ensures the redux store always has an activeSite value.
+ * It uses the URL as the source of truth and assumes the
+ * `site-slug` and `storage` query args are always set.
+ */
+export function EnsurePlaygroundSiteIsSelected({
+	children,
+}: {
+	children: React.ReactNode;
+}) {
+	const activeSite = useAppSelector((state) => state.activeSite);
+	const dispatch = useAppDispatch();
+	const [query] = useSearchParams();
+	const urlString = useCurrentUrl();
+	const requestedSiteSlug = query.get('site-slug');
+	const storage = query.get('storage')! as SiteStorageType;
+
+	useEffect(() => {
+		async function ensureSiteIsSelected() {
+			if (activeSite && activeSite.slug === requestedSiteSlug) {
+				if (activeSite.storage !== 'opfs') {
+					alert(
+						'Site slugs only work with browser storage. The site slug will be ignored.'
+					);
+				}
+				return;
+			}
+
+			console.log({ requestedSiteSlug });
+			if (requestedSiteSlug !== 'create') {
+				const siteInfo = await getSiteInfoBySlug(requestedSiteSlug!);
+				console.log({ siteInfo });
+				dispatch(setActiveSite(siteInfo!));
```

Comment:
What if the identified site does not exist?

One scenario:
1. User bookmarks Playground link for specific site slug/ID
2. User deletes site, perhaps unintentionally, or clears browser storage
3. User visits bookmark

Reply:
We need a notification system to tell them as we redirect to the first site that exists. It should be fine to delay that to another PR.